pkg/core/blockchain.go

```diff
@@ -535,8 +536,10 @@ func (bc *Blockchain) init() error {
 			return fmt.Errorf("failed to check native %s state against autogenerated one: %w", md.Name, err)
 		}
 		if !bytes.Equal(storedCSBytes, autogenCSBytes) {
-			return fmt.Errorf("native %s: version mismatch (stored contract state differs from autogenerated one), "+
-				"try to resynchronize the node from the genesis", md.Name)
+			storedJ, _ := json.Marshal(storedCS)
+			autogenJ, _ := json.Marshal(autogenCS)
+			return fmt.Errorf("native %s: version mismatch for the latest hardfork %s (stored contract state differs from autogenerated one), "+
+				"try to resynchronize the node from the genesis: %s vs %s", md.Name, current, string(storedJ), string(autogenJ))
```

Comment:
I'm not sure it's a friendly error message. Many users may even not notice the real message in these JSON dumps.

Reply:
These dumps are not indented, so they are quite compact, I've tested it. I may add some indent. Or is it better to remove dumps at all?